pyannote/audio/pipeline/speech_activity_detection.py

```diff
@@ -141,5 +150,26 @@ def __call__(self, current_file: dict) -> Annotation:
         return speech.to_annotation(generator='string', modality='speech')
 
     def get_metric(self) -> DetectionErrorRate:
-        """Return new instance of detection error rate metric"""
-        return  DetectionErrorRate(collar=0.0, skip_overlap=False)
+        """Return new instance of detection error rate metric
+        If mode set on detection.
+        Otherwise, it will use the self.loss function for precision and recall
+        """
+        if self.detection:
+            return DetectionErrorRate(collar=0.0, skip_overlap=False)
+        else:
+            raise NotImplementedError()
+
+    def loss(self, current_file: dict, hypothesis=None):
+        reference = current_file['annotation']
+        uem = get_annotated(current_file)
+
+        precision = DetectionPrecision()
+        recall = DetectionRecall()
+
+        p = precision(reference, hypothesis, uem=uem)
+        r = recall(reference, hypothesis, uem=uem)
+
+        if p > self.precision:
```

Comment:
We have a problem here. `self.precision` is not defined.

I propose that, instead of adding a `detection` parameter, we use a `precision` parameter that is set to None by default. None means "use detection error rate", any float value means "use precision/recall with this precision target".